### DISCUSSION TITLE
Should next.js handle the distribution of license information for open source dependencies?

### DISCUSSION THREAD

**Discussion by markjm** [vercel/next.js#33588] [Ideas]
### Describe the feature you'd like to request

After playing around with my create-next-app, I notice there is no builtin way to create and distribute the license files which are required to be available when using certain OSS. 

Generating such licenses is pretty easy with https://github.com/xz64/license-webpack-plugin (which NCC already uses). Given that information, I wonder if it makes sense to expose a "known route" to retrieve this information to make it easy to add a link to the notices.

In my opinion (and happy to discuss further) OSS compliance is exactly the "type" of web dev work that Next.js is equipped to totally handle and abstract away.

### Describe the solution you'd like

Perhaps use LicenseWebpackPlugin and expose the notice generated in some known way to be able to include it in the next.js documentation. Happy to help brainstorm further.

### Describe alternatives you've considered

Alternatives require users manually creating and exposing license information, but this goes against "zero config" ideals.

The more common alternative is for users to just ignore these licenses and not provide any mechanism to attribute the OSS they are using. (I should note that I was also unable to find this information for vercel.com)

**Comment by minecrawler** [comment]
This topic is a new pain point for me, because with the switch to Turbopack in V16, NextJS now offers no way to easily add license information to my artifacts anymore. I would love for NextJS to allow adding a banner text to its JS output files, so I can set up license information as legally required.
